### ISSUE TITLE
"Can not decrypt this file" for any file after upgrading from 7.0.0 to 8.1.3 and enabling encryption

### ISSUE THREAD

**Issue by gavv** [owncloud/core#19851]
Hello. I had owncloud 7.0.0 installation. Then I did this:

* removed all files in `/var/www/owncloud`, except `data/` and `config/`;
* unpacked `owncloud-latest.tar.bz2` there (8.1.3);
* pressed `start upgrade` button on web ui;

After that, I was unable to open any file in web ui. IIRC, the error message was this one:

> Installation is in transit between the old Encryption (ownCloud <= 8.0) and the new encryption.
Please enable the "Default encryption module" and run 'occ encryption:migrate'

Then I did this:

* enabled ` Default encryption module 1.0.0` app;
* enabled `Server-side encryption` option in Admin tab;
* executed `php occ encryption:migrate` (it reported success).

Now I'm still unable to open *any* file in web ui. The error is:

> Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.

This happens for all files, and I'm the file owner.

Log:

```
{"reqId":"WOsIqsx8JRgXEMxtU3D2","remoteAddr":"172.17.0.77","app":"PHP","message":"fread(): Length parameter must be greater than 0 at \/var\/www\/owncloud\/apps\/files_external\/3rdparty\/icewind\/streams\/src\/Wrapper.php#67","level":3,"time":"2015-10-18T09:09:20+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files_texteditor\/ajax\/loadfile?filename=poll.c&dir=%2Fdevel%2Fsnippets%2Fblog%2Ffifo"}
{"reqId":"WOsIqsx8JRgXEMxtU3D2","remoteAddr":"172.17.0.77","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2015-10-18T09:09:20+00:00","method":"GET","url":"\/owncloud\/index.php\/apps\/files_texteditor\/ajax\/loadfile?filename=poll.c&dir=%2Fdevel%2Fsnippets%2Fblog%2Ffifo"}
```

This is current directory structure:

```
.
|-- encryption_migration_backup_2015-10-18_07-53-57
|   `-- files_encryption
|-- encryption_migration_backup_2015-10-18_07-54-44
|   `-- files_encryption
|       `-- OC_DEFAULT_MODULE
|-- encryption_migration_backup_2015-10-18_07-56-30
|   `-- files_encryption
|       `-- OC_DEFAULT_MODULE
|           |-- pubShare_27d7d6cc.privateKey
|           `-- pubShare_27d7d6cc.publicKey
|-- files_encryption
|   `-- OC_DEFAULT_MODULE
|       |-- pubShare_27d7d6cc.privateKey
|       `-- pubShare_27d7d6cc.publicKey
`-- victor
    |-- encryption_migration_backup_2015-10-18_07-54-01
    |   |-- keyfiles
    |   |   |-- ...
    |   `-- victor.private.key
    |-- encryption_migration_backup_2015-10-18_07-54-45
    |   `-- victor.private.key
    |-- encryption_migration_backup_2015-10-18_07-56-30
    |   |-- OC_DEFAULT_MODULE
    |   |   |-- victor.privateKey
    |   |   `-- victor.publicKey
    |   `-- victor.private.key
    |-- files
    |   |-- ...
    |-- files_encryption
    |   |-- OC_DEFAULT_MODULE
    |   |   |-- victor.privateKey
    |   |   `-- victor.publicKey
    |   `-- victor.private.key
    |-- ...
```

My username is `victor`. I noted that:
* backuped `victor/encryption_migration_backup_2015-10-18_07-54-01/keyfiles/` mirrors `victor/files/` tree and contains a key file for every data file;
* but current `victor/files_encryption/` contains only user's private and public keys.

Is this a bug? Can I restore my files?

**Comment by RealRancor** [comment]
Hi,

a direct upgrade from oC 7.0.0 to 8.1.3 is not supported as stated in the upgrading docs. Please revert to your backup and do the following upgrade path:

oC 7.0.0 -> 7.0.10 -> 8.0.8 -> 8.1.3

**Comment by gavv** [comment]
Thanks.

What backups do I need? I have one for owncloud installation, but unfortunately not for data and db. Is it possible to restore data in this case?

**Comment by karlitschek** [comment]
@schiesbn any idea?

**Comment by RealRancor** [comment]
@karlitschek Is this really a bug? I think the only bug here is that the user wasn't warned that going from oC7 to 8.1 is not supported but that is fixed with oC 8.2 (https://github.com/owncloud/core/pull/18684) and 8.1.4 (https://github.com/owncloud/core/pull/19149).

**Comment by karlitschek** [comment]
I didn't want to confirm that this is a bug when I tagged it as a bug. I only wanted to express that this needs attention and investigation. Probably not a classic bug. This is true :-)

**Comment by RealRancor** [comment]
Ah, sorry. :)

**Comment by schiesbn** [comment]
> What backups do I need? I have one for owncloud installation, but unfortunately not for data and db. Is it possible to restore data in this case?

I can't guarantee it but it depends on what already changed on your system. Generally I often go back and forth between onwCloud versions on my devel system and often it work but it is in no way supported for productive systems.

But I think at this point you have no other chance than to try it. I would suggest:

1. Make a complete backup of your current ownCloud, including the data folder and the db so that you can go back and re-try if necessary
2. Replace your ownCloud installation with 8.0.8 and  try to migrate the keys as suggested by @RealRancor 
3. If you have any error messages post them here
4. Login and check if it was successful
5. If everything is fine install ownCloud 8.1.3 and follow again the update instruction posted by @RealRancor 
6, Login and check if everything works.

**Comment by gavv** [comment]
@schiesbn, @RealRancor, thanks a lot! I finally got it working.

What I exactly did:

* replaced owncloud installation with 8.0.8;
* restored `files_encryption/` directories for all users from corresponding `encryption_migration_backup_*` directories (with `keyfiles/` subdirectories);
* run `php occ upgrade`;
* enabled server-side encryption application;
* run `php occ encryption:migrate`;

At this point I was able to read all files again.

* replaced owncloud installation with 8.1.3;
* run `php occ upgrade`;
* run `php occ encryption:migrate`.

And now I have working 8.0.3 installation and no data lost.
